galley/pkg/config/analysis/local/analyze.go

```diff
@@ -61,7 +62,7 @@ type SourceAnalyzer struct {
 
 // NewSourceAnalyzer creates a new SourceAnalyzer with no sources. Use the Add*Source methods to add sources in ascending precedence order,
 // then execute Analyze to perform the analysis
-func NewSourceAnalyzer(m *schema.Metadata, analyzer *analysis.CombinedAnalyzer, cr snapshotter.CollectionReporterFn, serviceDiscovery bool) *SourceAnalyzer {
+func NewSourceAnalyzer(m *schema.Metadata, analyzer *analysis.CombinedAnalyzer, namespace string, cr snapshotter.CollectionReporterFn, serviceDiscovery bool) *SourceAnalyzer {
```

Comment:
Should additional tests be added for the local analyzer on namespace filtering?

Reply:
Absolutely. Just added a test to cover this. (And fixed all the broken ones, cough cough.)